src/components/avatar.rs

```diff
@@ -175,6 +196,41 @@ impl Avatar {
         }
     }
 
+    fn set_custom_icon(&self, user: Option<&User>, chat: Option<&Chat>) {
+        let imp = self.imp();
+        let icon_name = if let Some(user) = user {
+            matches!(user.type_().0, UserType::Deleted).then_some("background-app-ghost-symbolic")
+        } else if let Some(chat) = chat {
+            // Saved messages chat
+            chat.is_own_chat().then_some("user-bookmarks-symbolic")
+        } else {
+            None
+        };
+
+        if let Some(icon) = icon_name {
+            imp.avatar.set_icon_name(Some(icon));
+        };
+
+        imp.avatar.set_show_initials(icon_name.is_none());
+    }
```

Comment:
This should be replaced with the following function (so we have 2 simple update functions)
```rust
    fn update_avatar(&self) {
        let imp = self.imp();

        imp.avatar.set_custom_image(gdk::Paintable::NONE);
        imp.avatar.set_icon_name(None);
        imp.avatar.set_show_initials(true);

        if let Some(item) = self.item() {
            if let Some(user) = item.downcast_ref::<User>() {
                if user.type_().0 == UserType::Deleted {
                    imp.avatar
                        .set_icon_name(Some("background-app-ghost-symbolic"));
                    imp.avatar.set_show_initials(false);
                } else {
                    self.load_image(user.avatar(), user.session());
                }
            } else if let Some(chat) = item.downcast_ref::<Chat>() {
                if chat.is_own_chat() {
                    imp.avatar.set_icon_name(Some("user-bookmarks-symbolic"));
                    imp.avatar.set_show_initials(false);
                } else {
                    self.load_image(chat.avatar(), chat.session());
                }
            }
        }
    }
```